text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+});
+```
+
+The required `groupName` and the optional `description` properties are the identifiers for this group.
+
+The `role` property refers to the IAM role whose permissions that the users in this group are able to assume.
+
+When several different groups are defined for the same user pool, they can be ordered to have a precendence. Precedence
+will determine which IAM role's permission will be available for a user when they are part of two different groups. See
+[CreateGroup
+API](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateGroup.html#CognitoUserPools-CreateGroup-request-Precedence)
+for more details.
+
+Use the `takesPrecendenceOver()` API to set the precedence of one group over the other.
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * description - empty
+> * precedence - `undefined`
+
+## Identity Pools (Federated Identity)
+
+Identity pools enables creation of unique identities for their users and federate them with identity providers. These
+identity providers can be a Cognito user pool, Facebook, Google, or any other SAML-based identity provider.
+With an identity pool, you can obtain temporary, limited-privilege AWS credentials to access other AWS services.
+
+Using the CDK, a new identity pool can be created as part of the stack using the construct's constructor. You must
+specify the `identityPoolName`. This is the quickest way to create an identity pool using all of the defaults -
+
+```ts
+new IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myawesomeapp-identitypool',
+});
+```
+
+Identity pools can be configured to allow unauthenticated identities to access the application. This can be specified
+by setting the `unauthenticatedIdentities` boolean property. See [the Cognito
+guide](https://docs.aws.amazon.com/cognito/latest/developerguide/identity-pools.html) to learn more about
+unauthenticated identities.
+
+Identity pools can be configured so that a user authenticating with it will go through a different workflow to bootstrap
+their credentials. There are two different flows for authentication with public providers - basic and enhanced. See [the
+authentication flow page on Cognito](https://docs.aws.amazon.com/cognito/latest/developerguide/authentication-flow.html)
+to learn more about them. Use the `authenticationFlow` property to configure this.
+
+This is the code snippet with these properties configured -
+
+```ts
+new IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myawesomeapp-identitypool',
+  allowUnauthenticated: true,
+  authenticationFlow: AuthenticationFlow.ENHANCED,
+});
+```
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * allowUnauthenticated: false
+> * authenticationFlow: `AuthenticationFlow.ENHANCED`
+
+### Identity Providers
+
+Identity pools can be configured with external providers for sign in. These can be from facebook, google, twitter as
```

Comment:
Is that the difference between UserPool and IdentityPool? Direct users or federated users?

Reply:
Not quite. User pool lets you manage a pool of users, while identity pool lets users obtain temporary AWS credentials. Identity pools can use either a 3P identity provider (facebook, google, amazon,...) as well as the user pool as an identity provider.